tests/integration-tests/tests/sqlite_authority_tests.rs

```diff
@@ -954,9 +954,13 @@ async fn test_zone_signing() {
         assert!(
             inner_results
                 .iter()
-                .filter(|r| r.record_type() == RecordType::RRSIG && r.name() == record.name())
-                .filter_map(Record::data)
-                .filter_map(RRSIG::try_borrow)
+                .filter_map(|r| {
+                    if r.record_type() == RecordType::RRSIG && r.name() == record.name() {
+                        r.data().and_then(RRSIG::try_borrow)
+                    } else {
+                        None
+                    }
```

Comment:
Maybe rewrite it like this? I think this makes the logic a bit clearer? Alternatively, with a tuple scrutinee that has the `r.name() == record.name()` as the second element...

```suggestion
                    match r.record_type() {
                        RecordType::RRSIG if r.name() == record.name() => r.data().and_then(RRSIG::try_borrow),
                        _ => None,
                    }
```

I definitely do think this is easier to understand  than `.filter().filter_map().filter_map()` (and faster too, although of course that doesn't matter in test contexts).

Reply:
I can get behind this. It would be interesting to see what the compiler does with this, but you might be right that it's faster with fewer function calls.